### DISCUSSION TITLE
What type of coordinates is used in iNav code, and why comments "x axis of accelerometer is pointing backwards"?

### DISCUSSION THREAD

**Discussion by lida2003** [iNavFlight/inav#8483] [Q&A] [answered]
Hi

I'm new to this code. And I have two quesitons here:

a) What's the body frame coordinates we use in code?  ENU, xEast-yNorth-zUp , am I right?
b) In function imuMahonyAHRSupdate(when we use COG), why comments "x axis of accelerometer is pointing backwards." what does this mean?

BTW, x axis of accelerometer is pointing to the head of aircraft, right?

![图片](https://user-images.githubusercontent.com/1296171/196881289-4b2285b2-93b4-4c20-abee-90f494f99f3d.png)

**Comment by lida2003** [comment]
Is ENU the body frame coordination system for iNav's code?

**Reply by breadoven** [reply]
Defined here https://github.com/iNavFlight/inav/wiki/Developer-info

**Reply by lida2003** [reply]
Thanks for the links.

BTW, it seems NEU Coordinate System(iNav) is quite different from the description from https://ww2.mathworks.cn/help/map/choose-a-3-d-coordinate-system.html.

And I also noticed that :

>The x axis is aligned with the vector to the north pole (tangent to meridians).
>The y axis points to the east side (tangent to parallels)
>The z axis points up from the center of the earth

It's NOT right-handed (RH) Cartesian coordinate systems.

**So Is it left-handed (LH) Cartesian coordinate systems used in iNav Code?**

---

![图片](https://user-images.githubusercontent.com/1296171/197364813-b6d1cd7b-75a1-465f-ad34-9944bdd5e9f8.png)


![图片](https://user-images.githubusercontent.com/1296171/196855448-a54f155f-400b-47ec-9013-573737a01504.png)

**Comment by lida2003** [comment]
Does betaFlight/cleanFlight/iNav use the same coordinate system (left-handed (LH) Cartesian NEU coordinate systems)?

PS: BF & iNav are cloned from cleanflight (Multiwii).

**Comment by lida2003** [comment, marked as answer]
Thanks guys, I might be know the answer. xEast-yNorth-zUp + yaw(reverse), details see below link: [BetaFlight深入传感设计之九：传感坐标系/机体坐标系/导航坐标系/经纬度坐标系](https://blog.csdn.net/lida2003/article/details/127610897)
